uast/npath.go

```diff
@@ -0,0 +1,225 @@
+package uast
+
+import (
+	"errors"
+)
+
+//CodeReference
+//https://pmd.github.io/pmd-5.7.0/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/codesize/NPathComplexityRule.html
+
+//NpathComplexity return a slice of int for each function in the tree
+func NpathComplexity(n *Node) ([]int, error) {
+	var funcs []*Node
+	var npath []int
+
+	if n.containsRol(FunctionDeclarationBody) {
+		funcs = append(funcs, n)
+	} else {
+		funcs = n.deepChildrenOfRole(FunctionDeclarationBody)
+	}
+	if len(funcs) <= 0 {
```

Comment:
I'd find `len(funcs) == 0` less confusing, since `len()` return cannot be negative (no idea why `len()`'s return type is signed)